### PR TITLE
Merging `features/custom-fixed` to dev15.7.x

### PR DESCRIPTION
**Pull request by VSadov**
Test plan for "custom fixed": https://github.com/dotnet/roslyn/issues/25601
Test plan for "indexing movable fixed buffers": https://github.com/dotnet/roslyn/issues/25605

### CODE REVIEW

File: src/Compilers/CSharp/Test/Emit/CodeGen/FixedSizeBufferTests.cs [hunk 1]
```diff
@@ -40,38 +40,311 @@ static void Main()
...
+  IL_0018:  ldflda     ""S S1.field""
+  IL_001d:  ldflda     ""int* S.x""
+  IL_0022:  ldflda     ""int S.<x>e__FixedBuffer.FixedElementField""
+  IL_0027:  ldind.i4
+  IL_0028:  call       ""void System.Console.WriteLine(int)""
+  IL_002d:  ret
+}");
+        }
+
+        //PROTOTYPE: I have verified that this passes with the fix in https://github.com/dotnet/roslyn/pull/24909 that has not yet merged
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Emit/CodeGen/FixedSizeBufferTests.cs]: >PROTOTYPE [](start = 10, length = 9)

PROTOTYPE markers need to be addressed in a way or another before merging.

**Comment by VSadov** [comment]
@dotnet-bot  test windows_release_unit64_prtest please

**Comment by VSadov** [comment]
@dotnet-bot test windows_debug_unit32_prtest  please

**Comment by VSadov** [comment]
@dotnet-bot test windows_debug_unit32_prtest please

**Comment by VSadov** [comment]
@dotnet-bot test windows_debug_unit32_prtest please

**Comment by VSadov** [comment]
test windows_debug_vs-integration_prtest please

**Comment by VSadov** [comment]
test windows_release_vs-integration_prtest please

**Comment by VSadov** [comment]
test windows_debug_es_unit32_prtest please

**Comment by VSadov** [comment]
Thanks!!

**Pull request merged**